staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -148,6 +195,19 @@ func (i *indexedWatchers) deleteWatcher(number int, value string, supported bool
 	}
 }
 
+func (i *indexedWatchers) popWatchers(id int64) []*cacheWatcher {
```

Comment:
Shouldn't we also modify deleteWatcher - if we delete a watcher, shouldn't we also remove it from the "bookmark list"?

Reply:
this will be removed at timeout

Reply:
I realized it later in the code, but see this comment which remains valid: https://github.com/kubernetes/kubernetes/pull/75474/files#r272900754

Reply:
If there are many short watchers with large timeout, it will be a problem.